frontend/src/pages/projects/screens/spawner/connections/ConnectionsFormSection.tsx

```diff
@@ -0,0 +1,239 @@
+import ConnectionsTableRow from '~/pages/projects/screens/detail/connections/ConnectionsTableRow';
+import { SelectConnectionsModal } from './SelectConnectionsModal';
+import { connectionEnvVarConflicts, DuplicateEnvVarWarning } from './DuplicateEnvVarsWarning';
+import { DetachConnectionModal } from './DetachConnectionModal';
+
+const columns: SortableData<Connection>[] = [
+  {
+    field: 'name',
+    label: 'Name',
+    sortable: (a, b) =>
+      getDisplayNameFromK8sResource(a).localeCompare(getDisplayNameFromK8sResource(b)),
+  },
+  {
+    field: 'type',
+    label: 'Type',
+    sortable: (a, b) =>
+      a.metadata.annotations['opendatahub.io/connection-type'].localeCompare(
+        b.metadata.annotations['opendatahub.io/connection-type'],
+      ),
+  },
+  {
+    field: 'kebab',
+    label: '',
+    sortable: false,
+  },
+];
+
+type Props = {
+  project: ProjectKind;
+  projectConnections: Connection[];
+  refreshProjectConnections: () => void;
+  notebook?: NotebookKind;
+  notebookDisplayName: string;
+  selectedConnections: Connection[];
+  setSelectedConnections: (connections: Connection[]) => void;
+};
+
+export const ConnectionsFormSection: React.FC<Props> = ({
+  project,
+  projectConnections,
+  refreshProjectConnections,
+  notebook,
+  notebookDisplayName,
+  selectedConnections,
+  setSelectedConnections,
+}) => {
+  const [connectionTypes] = useWatchConnectionTypes();
+
+  const [initialNumberConnections] = React.useState(selectedConnections.length);
+  const notebookArray = React.useMemo(() => (notebook ? [notebook] : []), [notebook]);
+  const [notebookStates] = useNotebooksStates(
+    notebookArray,
+    notebook?.metadata.namespace || '',
+    initialNumberConnections > 0,
+  );
+  const isRunning = React.useMemo(
+    () =>
+      !!notebookStates.find((n) => n.notebook.metadata.name === notebook?.metadata.name)?.isRunning,
+    [notebookStates, notebook],
+  );
+
+  const unselectedConnections = React.useMemo(
+    () =>
+      projectConnections.filter(
+        (pc) => !selectedConnections.find((sc) => pc.metadata.name === sc.metadata.name),
+      ),
+    [projectConnections, selectedConnections],
+  );
+
+  const [showAttachConnectionsModal, setShowAttachConnectionsModal] = React.useState(false);
+  const [detachConnectionModal, setDetachConnectionModal] = React.useState<Connection>();
+  const [manageConnectionModal, setManageConnectionModal] = React.useState<{
+    connection?: Connection;
+    isEdit?: boolean;
+  }>();
+
+  const envVarConflicts = React.useMemo(
+    () => connectionEnvVarConflicts(selectedConnections),
+    [selectedConnections],
+  );
+
+  return (
+    <FormSection
+      title={
+        <>
+          {SpawnerPageSectionTitles[SpawnerPageSectionID.CONNECTIONS]}{' '}
+          <Tooltip
+            content="No existing connections available"
+            trigger={unselectedConnections.length === 0 ? 'mouseenter focus' : 'manual'}
```

Comment:
This is a bit deceptive. I'd rather conditionally add the tooltip than have it added but not be able to be triggered.